findspark.py

```diff
@@ -35,7 +36,81 @@ def find():
     return spark_home
 
 
-def init(spark_home=None):
+def change_rc(spark_home, spark_python, py4j):
+    """Persists changes to enviornment by changing shell config.
+
+    Adds lines to .bashrc to set enviornment variables including
+    the adding of dependencies to the system path. Currently only 
+    works for Bash and (t)csh.
+
+    Parameters
+    ----------
+    spark_home : str
+        Path to Spark installation.
+    spark_python : str
+        Path to python subdirectory of Spark installation.
+    py4j : str
+        Path to py4j library.
+    """
+
+    bashrc_location = os.path.expanduser("~/.bashrc")
+
+    with open(bashrc_location, 'a') as bashrc:
+        bashrc.write("\n# Added by findspark\n")
+        bashrc.write("export SPARK_HOME=" + spark_home + "\n")
+        bashrc.write("export PYTHONPATH=" + spark_python + ":" + 
+                     py4j + ":$PYTHONPATH\n\n")
+    
+    cshrc_location = os.path.expanduser("~/.cshrc")
+    
+    with open(cshrc_location, 'a') as cshrc:
+        cshrc.write("\n# Added by findspark\n")
+        cshrc.write("setenv SPARK_HOME " + spark_home + "\n")
+        cshrc.write("setenv PYTHONPATH \"" + spark_python + ":" +
+                    py4j + ":\"$PYTHONPATH")
+ 
+
+def edit_ipython_profile(spark_home, spark_python, py4j, name):
+    """Creates or appends to an IPython profile to automatically import pyspark.
+    
+    Adds lines to the ipython_config file to be run at startup of the IPython
+    interpreter for a given profile, creating the profile if it does not exist.
+    These lines set appropriate enviornment variables and import the pyspark 
+    library upon IPython startup.
+
+    Parameters
+    ----------
+    spark_home : str
+        Path to Spark installation.
+    spark_python : str
+        Path to python subdirectory of Spark installation.
+    py4j : str
+        Path to py4j library.
+    name : str
+        Name of profile to create or append to.
+    """
+    subprocess.call(["ipython", "profile", "create", name])
```

Comment:
A few things here:

1. let's store it in the current IPython profile, rather than creating a new one.
2. Create a startup file, rather than editing a config file. The file can be `<PROFILE>/startup/findspark.py`.
3. There's no need to call out to subprocesses. We can get the location with a Python API:

```python
from IPython import get_ipython
ip = get_ipython()
if ip:
    profile_dir = ip.profile_dir.location
else:
    from IPython.utils.path import locate_profile
    profile_dir = locate_profile()

startup_file = os.path.join(profile_dir, 'startup', 'findspark.py')
```